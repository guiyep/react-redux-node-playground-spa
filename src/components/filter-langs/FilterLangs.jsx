import React, { memo, useCallback } from 'react';
import PropTypes from 'prop-types';
import { AutoComplete } from 'antd';

const FilterLangs = memo(({ onChange, defaultSelection, options }) => {
  const onChangeHandler = useCallback(
    value => {
      if (options.includes(value)) {
        onChange(value);
      }
    },
    [onChange, options],
  );

  return (
    <AutoComplete
      dataSource={options}
      defaultValue={defaultSelection}
      onChange={onChangeHandler}
      placeholder="Select a language"
    />
  );
});

FilterLangs.displayName = 'FilterLangs';

FilterLangs.propTypes = {
  onChange: PropTypes.func.isRequired,
  options: PropTypes.array.isRequired,
  defaultSelection: PropTypes.string,
};

FilterLangs.defaultProps = {
  defaultSelection: undefined,
};

export default FilterLangs;
