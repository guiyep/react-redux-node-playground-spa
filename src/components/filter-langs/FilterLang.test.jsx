import React from 'react';
import { create } from 'react-test-renderer';
import FilterLangs from './FilterLangs';
import ReactDOM from 'react-dom';

const options = ['test', '1', '2', '3'];

describe('FilterLangs', () => {
  test('it matches the snapshot', () => {
    const component = create(<FilterLangs options={options} onChange={() => {}} defaultSelection="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<FilterLangs options={options} onChange={() => {}} defaultSelection="test" />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
