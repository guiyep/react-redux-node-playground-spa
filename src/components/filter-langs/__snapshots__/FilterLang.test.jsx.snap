// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterLangs it matches the snapshot 1`] = `
<div
  className="ant-select-show-search ant-select-auto-complete ant-select ant-select-combobox ant-select-enabled"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseOut={[Function]}
  onMouseUp={[Function]}
  onTouchStart={[Function]}
>
  <div
    aria-autocomplete="list"
    aria-controls="test-uuid"
    aria-expanded={false}
    aria-haspopup="true"
    className="ant-select-selection
            ant-select-selection--single"
    role="combobox"
  >
    <div
      className="ant-select-selection__rendered"
    >
      <div
        className="ant-select-selection__placeholder"
        onClick={[Function]}
        onMouseDown={[Function]}
        style={
          Object {
            "WebkitUserSelect": "none",
            "display": "none",
            "userSelect": "none",
          }
        }
        unselectable="on"
      >
        Select a language
      </div>
      <ul>
        <li
          className="ant-select-search ant-select-search--inline"
        >
          <div
            className="ant-select-search__field__wrap"
          >
            <input
              className="ant-input ant-select-search__field"
              disabled={false}
              onChange={[Function]}
              onKeyDown={[Function]}
              type="text"
              value="test"
            />
            <span
              className="ant-select-search__field__mirror"
            >
              test
              Â 
            </span>
          </div>
        </li>
      </ul>
    </div>
    <span
      className="ant-select-arrow"
      onClick={[Function]}
      style={
        Object {
          "WebkitUserSelect": "none",
          "userSelect": "none",
        }
      }
      unselectable="on"
    >
      <i
        aria-label="icon: down"
        className="anticon anticon-down ant-select-arrow-icon"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="down"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
          />
        </svg>
      </i>
    </span>
  </div>
</div>
`;
